### PR TITLE
MCR-1807 fixed session handling

### PR DESCRIPTION
**Pull request by sebhofmann**
* MCRSessionManager can now lock a thread. So no session can be created.
* Use mycore default jersey configuration in RESTAPI
* Created filter which can lock session creation
* Use filter when MCRStatic is annotated
* Removed MCRStatic from RESTAPI because it uses sessions

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 3dde7aa2abe6fc02e244256ddf077d40abf27ba3! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/2804-pull_request-517

**Review by rsteph-de — COMMENTED** [review]
MCRSessionMgr.lock() and unlock()
increase the complexity of MCR session handling

File: mycore-restapi/src/main/java/org/mycore/restapi/MCRRESTResourceConfig.java [hunk 1]
```diff
@@ -3,26 +3,33 @@
...
-        String[] restPackages = MCRConfiguration.instance().getStrings("MCR.RestAPI.Resource.Packages").stream()
-            .toArray(String[]::new);
-        packages(restPackages);
-        register(MultiPartFeature.class);
         register(MCRRestAPIExceptionMapper.class);
     }
 
+    @Override
+    protected void setupResources() {
+        String propertyString = MCRConfiguration.instance().getString("MCR.RestAPI.Resource.Packages");
```
Comment (yagee-de) [inline mycore-restapi/src/main/java/org/mycore/restapi/MCRRESTResourceConfig.java]: use getStrings() here

File: mycore-restapi/src/main/java/org/mycore/restapi/feature/MCRRESTFeature.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.mycore.common.config.MCRConfiguration;
+import org.mycore.frontend.jersey.feature.MCRJerseyDefaultFeature;
+import org.mycore.restapi.MCRRESTResourceConfig;
+
+public class MCRRESTFeature extends MCRJerseyDefaultFeature {
+
+    @Override
+    protected List<String> getPackages() {
+        String propertyString = MCRConfiguration.instance().getString(MCRRESTResourceConfig.REST_API_PACKAGE);
```
Comment (yagee-de) [inline mycore-restapi/src/main/java/org/mycore/restapi/feature/MCRRESTFeature.java]: use getStrings() here

**Review by yagee-de — CHANGES_REQUESTED** [review]
two minor changes requested. Rest looks OK to me.

**Review by Mewel — APPROVED** [review]
works with jportal

**Review by kkrebs — CHANGES_REQUESTED** [review]
Tested in MIR. There is still one session type staying open.

**Review by yagee-de — APPROVED** [review]

**Review by kkrebs — APPROVED** [review]

**Pull request merged**